helm install <release-name> does not work unless release-name == 'projectriff'

Using the 0.0.6-snapshot helm chart on Apr 03, 2018

Would it be possible to parameterize the kafka broker and zookeeper values to use the release-name instead of hardwiring?

This works
```
helm install --name projectriff --namespace riff-system projectriff/riff --devel --set rbac.create=false --set kafka.create=true --set httpGateway.service.type=NodePort
```

This doesn't
```
helm install --name control --namespace riff-system projectriff/riff --devel --set rbac.create=false --set kafka.create=true --set httpGateway.service.type=NodePort
```

topic-controller log
```
panic: lookup projectriff-zookeeper.riff-system on 10.96.0.10:53: no such host
 goroutine 1 [running]:
github.com/projectriff/riff/topic-controller/pkg/topic/provisioner/kafka.NewKafkaProvisioner(0xc42001404f, 0x26, 0xc42001404f, 0x26)
	/go/src/github.com/projectriff/riff/topic-controller/pkg/topic/provisioner/kafka/kafka.go:32 +0xd0
main.main()
	/go/src/github.com/projectriff/riff/topic-controller/cmd/topic-controller.go:53 +0xd1
```

gateway log
```

panic: kafka: client has run out of available brokers to talk to (Is your cluster reachable?)
 goroutine 1 [running]:
main.main()
	/go/src/github.com/projectriff/riff/http-gateway/cmd/http-gateway.go:37 +0x3fe
```

--- a/helm-charts/riff/templates/_helpers.tpl
+++ b/helm-charts/riff/templates/_helpers.tpl
@@ -22,7 +22,7 @@ Create the list of kafka broker nodes to use
 {{- if .Values.kafka.create -}}
     {{ default ( printf "%s-%s.%s:9092" .Release.Name "kafka" .Release.Namespace ) .Values.kafka.broker.nodes }}
 {{- else -}}
-    {{ .Values.kafka.broker.nodes }}
+    {{ required "A valid .Values.kafka.broker.nodes entry required!" .Values.kafka.broker.nodes }}
 {{- end -}}
 {{- end -}}
 
@@ -33,7 +33,7 @@ Create the list of kafka zookeeper nodes to use
 {{- if .Values.kafka.create -}}
     {{ default ( printf "%s-%s.%s:2181" .Release.Name "zookeeper" .Release.Namespace ) .Values.kafka.zookeeper.nodes }}
 {{- else -}}
-    {{ .Values.kafka.zookeeper.nodes }}
+    {{ required "A valid .Values.kafka.zookeeper.nodes entry required!" .Values.kafka.zookeeper.nodes }}
 {{- end -}}
 {{- end -}}
 

--- a/helm-charts/riff/templates/_helpers.tpl
+++ b/helm-charts/riff/templates/_helpers.tpl
@@ -16,6 +16,28 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this 
 {{- end -}}
 
 {{/*
+Create the list of kafka broker nodes to use
+*/}}
+{{- define "riff.kafkaBrokers" -}}
+{{- if .Values.kafka.create -}}
+    {{ default ( printf "%s-%s.%s:9092" .Release.Name "kafka" .Release.Namespace ) .Values.kafka.broker.nodes }}
+{{- else -}}
+    {{ .Values.kafka.broker.nodes }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Create the list of kafka zookeeper nodes to use
+*/}}
+{{- define "riff.kafkaZkNodes" -}}
+{{- if .Values.kafka.create -}}
+    {{ default ( printf "%s-%s.%s:2181" .Release.Name "zookeeper" .Release.Namespace ) .Values.kafka.zookeeper.nodes }}
+{{- else -}}
+    {{ .Values.kafka.zookeeper.nodes }}
+{{- end -}}
+{{- end -}}
+
+{{/*
 Create the name of the service account to use
 */}}
 {{- define "riff.serviceAccountName" -}}

--- a/helm-charts/riff/templates/function-controller-deployment.yaml
+++ b/helm-charts/riff/templates/function-controller-deployment.yaml
@@ -37,7 +37,7 @@ spec:
             - containerPort: 8080
           env:
           - name: KAFKA_BROKERS
-            value: {{ .Values.kafka.broker.nodes }}
+            value: {{ template "riff.kafkaBrokers" . }}
           - name: RIFF_FUNCTION_SIDECAR_REPOSITORY
             value: {{ .Values.functionController.sidecar.image.repository }}
           - name: RIFF_FUNCTION_SIDECAR_TAG

--- a/helm-charts/riff/templates/http-gateway-deployment.yaml
+++ b/helm-charts/riff/templates/http-gateway-deployment.yaml
@@ -37,7 +37,7 @@ spec:
             - containerPort: 8080
           env:
           - name: KAFKA_BROKERS
-            value: {{ .Values.kafka.broker.nodes }}
+            value: {{ template "riff.kafkaBrokers" . }}
           - name: RIFF_HTTP_HEADERS_WHITELIST
             value: {{ .Values.httpGateway.httpHeadersWhitelist }}
       serviceAccountName: {{ template "riff.serviceAccountName" . }}

--- a/helm-charts/riff/templates/topic-controller-deployment.yaml
+++ b/helm-charts/riff/templates/topic-controller-deployment.yaml
@@ -37,6 +37,6 @@ spec:
             - containerPort: 8080
           env:
           - name: KAFKA_ZK_NODES
-            value: {{ .Values.kafka.zookeeper.nodes }}
+            value: {{ template "riff.kafkaZkNodes" . }}
       serviceAccountName: {{ template "riff.serviceAccountName" . }}
 {{- end -}}

--- a/helm-charts/riff/values.yaml
+++ b/helm-charts/riff/values.yaml
@@ -51,6 +51,6 @@ kafka:
   # Specifies whether the single-node development Kafka chart should be installed
   create: false
   broker:
-    nodes: projectriff-kafka.riff-system:9092
+    nodes: null
   zookeeper:
-    nodes: projectriff-zookeeper.riff-system:2181
+    nodes: null

